Source/WebCore/platform/graphics/gstreamer/AudioTrackPrivateGStreamer.cpp

```diff
@@ -31,70 +31,93 @@
     , m_player(player)
 {
-    int kind;
-    auto tags = adoptGRef(gst_stream_get_tags(m_stream.get()));
-
-    if (tags && gst_tag_list_get_int(tags.get(), "webkit-media-stream-kind", &kind) && kind == static_cast<int>(AudioTrackPrivate::Kind::Main)) {
-        auto streamFlags = gst_stream_get_stream_flags(m_stream.get());
-        gst_stream_set_stream_flags(m_stream.get(), static_cast<GstStreamFlags>(streamFlags | GST_STREAM_FLAG_SELECT));
-    }
-
+    ensureDebugCategoryInitialized();
     g_signal_connect_swapped(m_stream.get(), "notify::caps", G_CALLBACK(+[](AudioTrackPrivateGStreamer* track) {
         track->m_taskQueue.enqueueTask([track]() {
-            track->updateConfigurationFromCaps();
+            auto caps = adoptGRef(gst_stream_get_caps(track->m_stream.get()));
+            track->capsChanged(String::fromLatin1(gst_stream_get_stream_id(track->m_stream.get())), caps);
         });
     }), this);
     g_signal_connect_swapped(m_stream.get(), "notify::tags", G_CALLBACK(+[](AudioTrackPrivateGStreamer* track) {
         track->m_taskQueue.enqueueTask([track]() {
-            track->updateConfigurationFromTags();
+            auto tags = adoptGRef(gst_stream_get_tags(track->m_stream.get()));
+            track->updateConfigurationFromTags(tags);
         });
     }), this);
 
-    updateConfigurationFromCaps();
-    updateConfigurationFromTags();
+    auto caps = adoptGRef(gst_stream_get_caps(m_stream.get()));
+    updateConfigurationFromCaps(caps);
+
+    auto tags = adoptGRef(gst_stream_get_tags(m_stream.get()));
+    updateConfigurationFromTags(tags);
 }
 
-void AudioTrackPrivateGStreamer::updateConfigurationFromTags()
+void AudioTrackPrivateGStreamer::capsChanged(const String& streamId, const GRefPtr<GstCaps>& caps)
 {
     ASSERT(isMainThread());
-    if (!m_stream)
-        return;
+    updateConfigurationFromCaps(caps);
 
-    auto tags = adoptGRef(gst_stream_get_tags(m_stream.get()));
-    unsigned bitrate;
-    if (!tags || !gst_tag_list_get_uint(tags.get(), GST_TAG_BITRATE, &bitrate))
+    auto codec = m_player->codecForStreamId(streamId);
+    if (codec.isEmpty())
         return;
-
+    GST_DEBUG_OBJECT(objectForLogging(), "Setting codec to %s", codec.ascii().data());
     auto configuration = this->configuration();
-    configuration.bitrate = bitrate;
+    configuration.codec = WTFMove(codec);
     setConfiguration(WTFMove(configuration));
 }
 
-void AudioTrackPrivateGStreamer::updateConfigurationFromCaps()
+void AudioTrackPrivateGStreamer::updateConfigurationFromTags(const GRefPtr<GstTagList>& tags)
 {
     ASSERT(isMainThread());
-    if (!m_stream)
+    GST_DEBUG_OBJECT(objectForLogging(), "Updating audio configuration from %" GST_PTR_FORMAT, tags.get());
+    if (!tags)
         return;
 
-    auto caps = adoptGRef(gst_stream_get_caps(m_stream.get()));
+    auto configuration = this->configuration();
+    auto scopeExit = makeScopeExit([&] {
```

Comment:
This is not needed, I think, as I see no other exit than function's end.

Reply:
indeed